src/jalv.c

```diff
@@ -75,14 +72,19 @@
 #    define MAX(a, b) (((a) > (b)) ? (a) : (b))
 #endif
 
+#ifndef ARRAY_SIZE
+#    define ARRAY_SIZE(arr) (sizeof(arr) / sizeof(arr[0]))
+#endif
+
 /* Size factor for UI ring buffers.  The ring size is a few times the size of
    an event output to give the UI a chance to keep up.  Experiments with Ingen,
    which can highly saturate its event output, led me to this value.  It
    really ought to be enough for anybody(TM).
 */
 #define N_BUFFER_CYCLES 16
 
-ZixSem exit_sem;  /**< Exit semaphore */
+/* only used for the signal handler */
+static ZixSem* exit_sem = NULL;  /**< Exit semaphore */
```

Comment:
The pointer will be read and write atomically. Therefore the signal handler will either see a NULL pointer or a valid pointer to the semaphore. Due to that I see no problems with accessing this pointer from the signal handler.
I will add `volatile` to ensure that the value is updated in time.

Do you see any other problems?